### PR TITLE
Added macOS target along with tests and examples

### PR DESCRIPTION
**Pull request by Adamcbrz**
#6

### CODE REVIEW

File: Examples/Hyperspace-macOSExample/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        alert.messageText = title
+        alert.informativeText = message
+        alert.alertStyle = .warning
+        
+        alert.runModal()
+    }
+}
+
+extension NSApplication: NetworkActivityIndicatable {
+    public var isNetworkActivityIndicatorVisible: Bool {
```
Comment (tylermilner) [inline Examples/Hyperspace-macOSExample/ViewController.swift]: We may want to consider making `isNetworkActivityIndicatorVisible` a `{ get }` only requirement on `NetworkActivityIndicatable` so that we don't have to have the no-op below.

Comment (Adamcbrz) [reply Examples/Hyperspace-macOSExample/ViewController.swift]: Looks like I don't need that and it can match the other examples.

File: Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift [hunk 2]
```diff
@@ -10,7 +10,11 @@ import Foundation
 
 /// Represents an object that can indicate to the user that network activity is currently taking place.
 public protocol NetworkActivityIndicatable {
+    #if MACOS
+    //macOS does not have a activity indicator
```
Comment (wmcginty) [inline Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift]: This doesn't need to be here - the point of abstracting UIApplication's functionality into a protocol is so a custom indicator could be built on any platform.

In the case of macOS, if the client doesn't want to create a custom indicator they can just pass nil into the NetworkService.

Having said that, we will probably need to remove the default value of UIApplication.shared.

File: Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift [hunk 3]
```diff
@@ -74,11 +78,15 @@ private extension NetworkActivityController {
         delayedHide?.cancel()
         delayedHide = nil
         
+        #if MACOS
```
Comment (wmcginty) [inline Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift]: Same as above - the protocol is abstracting the use of UIApplication. The isNetworkActivityIndicatorVisible call could be forwarded to a custom indicator (as it would need to be on macOS).

File: Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift (hunk 2 above)
Comment (tylermilner) [reply Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift]: As for removing the default value of `UIApplication.shared`, do you think we could use the #if macros to define two different function signatures? One that sets no default for macOS and then one that sets the default for everything else?

**Comment by tylermilner** [comment]
Just realized one more area that we need to update is the Readme too.

File: Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift (hunk 2 above)
Comment (wmcginty) [reply Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift]: We could either do that or create an extension on NetworkService that is only compiled into iOS/tvOS. The base initializer would have no default, the iOS/tvOS extension would have the default.


Names may need to be tweaked to avoid ambiguity, but I think that's a little cleaner than #if, because now that I think about it watchOS doesn't have UIApplication either.

Comment (tylermilner) [reply Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift]: Yeah, that sounds like a good approach.

**Comment by tylermilner** [comment]
And...just realized we'll also need to update `.travis.yml` too. 😅

File: .travis.yml [hunk 4]
```diff
@@ -37,5 +37,3 @@ script:
   - if [ $POD_LINT == "YES" ]; then
       pod lib lint;
     fi
-after_success:
```
Comment (wmcginty) [inline .travis.yml]: I'm guessing this a product of the merge - but we'll want to ensure this is present. I think we'll also need to ensure the macOS variant is building on CI.

File: README.md [hunk 5]
```diff
@@ -4,8 +4,6 @@
 [![Version](https://img.shields.io/cocoapods/v/Hyperspace.svg?style=flat)](http://cocoapods.org/pods/Hyperspace)
 [![License](https://img.shields.io/cocoapods/l/Hyperspace.svg?style=flat)](http://cocoapods.org/pods/Hyperspace)
 [![Platform](https://img.shields.io/cocoapods/p/Hyperspace.svg?style=flat)](http://cocoapods.org/pods/Hyperspace)
-[![codecov](https://codecov.io/gh/BottleRocketStudios/iOS-Hyperspace/branch/master/graph/badge.svg)](https://codecov.io/gh/BottleRocketStudios/iOS-Hyperspace)
```
Comment (wmcginty) [inline README.md]: Same again - I think it's a product of the merge. But we'll want both of these badges to remain present in the Readme

File: .travis.yml (hunk 4 above)
Comment (wmcginty) [reply .travis.yml]: I'm getting my branches mixed up. Ignore me.

File: README.md (hunk 5 above)
Comment (wmcginty) [reply README.md]: I'm getting my branches mixed up. Ignore me.

File: .travis.yml (hunk 4 above)
Comment (tylermilner) [reply .travis.yml]: @wmcginty I guess I'm suffering from the same mixup. It looks like this is merging into 'master', which already has the codecov stuff so don't we need to get this fixed?

Comment (wmcginty) [reply .travis.yml]: @tylermilner On third viewing, you're right. Those changes were merged into master in commit db315e so these do need to be corrected.

Comment (Adamcbrz) [reply .travis.yml]: done

File: README.md (hunk 5 above)
Comment (Adamcbrz) [reply README.md]: Added both codecov & codebeat from db315ed don't know why they didn't merge in properly.

**Comment by wmcginty** [comment]
@tylermilner are you still looking for changes on this? It looks like the Readme changes have been fixed.

File: Sources/Hyperspace/Service/Backend/BackendServiceProtocol.swift [hunk 6]
```diff
@@ -87,6 +87,6 @@ extension BackendServiceError: Equatable {
 
 extension AnyError: BackendServiceErrorInitializable {
     public init(_ backendServiceError: BackendServiceError) {
-        self.init(backendServiceError as Error)
+        self.init(backendServiceError)
```
Comment (tylermilner) [inline Sources/Hyperspace/Service/Backend/BackendServiceProtocol.swift]: @Adamcbrz Was this change intentional?

File: Tests/BackendServiceTests.swift [hunk 7]
```diff
@@ -111,22 +111,4 @@ class BackendServiceTests: XCTestCase {
...
-        let rhs = BackendServiceError.dataTransformationError(NSError(domain: "Test", code: 1, userInfo: nil))
-        XCTAssertFalse(lhs == rhs)
-    }
-    
-    func test_BackendServiceProtocol_Execute() {
-        let mock = MockBackendService()
-        mock.execute(request: NetworkRequestTests.SimpleGETRequest()) { (result: Result<NetworkRequestTests.SimpleGETRequest.ResponseType, NetworkRequestTests.SimpleGETRequest.ErrorType>) in
-            XCTAssert(true)
-        }
-    }
```
Comment (tylermilner) [inline Tests/BackendServiceTests.swift]: It looks like this is causing a ~4% decrease in code coverage. Was it intentional to remove these tests?

File: Tests/NetworkRequestTests.swift [hunk 8]
```diff
@@ -109,39 +84,6 @@ class NetworkRequestTests: XCTestCase {
...
-        XCTAssertNotNil(result.value)
-    }
-    
-    func test_NetworkRequest_CustomQueryEncoding() {
-        let request = CustomQueryEncodingRequest()
-        let queryString = request.encodeQueryParameterString("this is a test")
-        
-        XCTAssert(queryString == "this-is-a-test")
-    }
-
```
Comment (tylermilner) [inline Tests/NetworkRequestTests.swift]: Same question for these tests.

File: Tests/NetworkServiceTests.swift [hunk 9]
```diff
@@ -122,17 +122,6 @@ class NetworkServiceTests: XCTestCase {
...
-        let networkService = NetworkServiceMockSubclass(session: URLSession.shared, networkActivityIndicatable: MockNetworkActivityIndicator())
-        
-        XCTAssert(networkService.initWithSessionNetworkActivityControllerWasCalled)
-    }
-    
-    func test_NetworkServiceHelper_InvalidHTTPResponsErrorUnknownError() {
-        let networkServiceFailure = NetworkServiceHelper.networkServiceFailure(for: NSError(domain: NSURLErrorDomain, code: NSURLErrorBadURL, userInfo: nil))
-        XCTAssert(networkServiceFailure.error == .unknownError)
-    }
-
```
Comment (tylermilner) [inline Tests/NetworkServiceTests.swift]: These as well.

**Comment by tylermilner** [comment]
@wmcginty I noticed that there was some code removed that didn't look intentional. I assumed there were still changes to be made to correct them. Added explicit comments for them now.

**Comment by tylermilner** [comment]
I think the .travis.yml still needs updating as well.

**Comment by wmcginty** [comment]
Sounds good- just wanted to make sure these changes didn't fall through the cracks.

**Comment by wmcginty** [comment]
Going to close this due to inactivity. We also now have macOS support after #151 .
